spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/route/Route.java

```diff
@@ -114,7 +115,11 @@ public B uri(String uri) {
 
 		public B uri(URI uri) {
 			this.uri = uri;
-			if (this.uri.getPort() < 0 && this.uri.getScheme().startsWith("http")) {
+			String scheme = this.uri.getScheme();
+			if (StringUtils.isEmpty(scheme)) {
+				throw new IllegalArgumentException("The parameter [" + this.uri + "] format is incorrect ");
```

Comment:
Be specific, say that the scheme is missing.

Reply:
Can you look at this [question](https://stackoverflow.com/questions/51148121/file-scheme-for-route-uri-field)?